### DISCUSSION TITLE
How to specify different compilers per dependency in manifest mode ?

### DISCUSSION THREAD

**Discussion by abdelmaged** [microsoft/vcpkg#30766] [Q&A]
Let's say I have two dependencies libX & libY.
I want to compile libX with compiler gcc-5.4.0
and compile libY with compiler gcc-10.2.0

assuming same platform (Linux) and same architecture (x64)

What is the best way to achieve this in manifest mode ?

**Comment by autoantwort** [comment]
You could use https://learn.microsoft.com/en-us/vcpkg/users/triplets#per-port-customization and [use different toolchains](https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_chainload_toolchain_file) for the libs.

**Comment by autoantwort** [comment]
Why do you want to do that?

**Reply by abdelmaged** [reply]
I know it looks weird, but I'm migrating old system, and it turns out it uses different compiler versions for some third party libraries.

**Reply by autoantwort** [reply]
Then maybe now is the time to unify the compiler versions and don't mimic wired old behavior.

**Reply by abdelmaged** [reply]
I hope I can do that, but updating the compiler is not my decision (or even my team), it's complicated with some internal upstream/downstream projects in different teams which some of them is legacy code and it won't be easy task to update the code to compile with newer compiler versions, this is besides the time needed to discuss and conclude with different stakeholders about this.

**Reply by plimy** [reply]
I can give you a real life use case. I have an embedded system with multiple processors, a total of 3 different architectures. My project needs to build a binary image for each processor, and then package them into a .zds. Several of these processors use the same libraries (compiled with a different cross compiler).

I am contemplating moving the package manager to vcpkg, because the current method my team is using is not scaling well... and vcpkg looks like it would be much easier to use then our current package management system... If it can handle different compilers.

**Reply by autoantwort** [reply]
So you have one single cmake project that builds the images for three different architectures in one go?

**Reply by plimy** [reply]
Not exactly, its more like one top level project with 3 sub-projects that each use a different toolchain.... Its not particularly different for using Cmake for multi-language projects.
